OrbitGl/FrameTrackOnlineProcessor.cpp

```diff
@@ -0,0 +1,50 @@
+#include "FrameTrackOnlineProcessor.h"
+
+#include <limits>
+#include "CaptureData.h"
+#include "TimeGraph.h"
+#include "UserDefinedCaptureData.h"
+
+FrameTrackOnlineProcessor::FrameTrackOnlineProcessor(const CaptureData& capture_data,
+                                                     TimeGraph* time_graph)
+    : time_graph_(time_graph) {
+  const UserDefinedCaptureData& user_defined_capture_data =
+      capture_data.user_defined_capture_data();
+  for (const auto& function : user_defined_capture_data.frame_track_functions()) {
+    const uint64_t function_address = capture_data.GetAbsoluteAddress(function);
+    current_frame_track_functions_.insert(function_address);
+    previous_timestamp_ns_.insert(
+        std::make_pair(function_address, std::numeric_limits<uint64_t>::max()));
+  }
+}
+
+void FrameTrackOnlineProcessor::ProcessTimer(const orbit_client_protos::TimerInfo& timer_info,
+                                             const orbit_client_protos::FunctionInfo& function) {
+  uint64_t function_address = timer_info.function_address();
+  if (current_frame_track_functions_.empty() ||
+      !current_frame_track_functions_.contains(function_address)) {
+    return;
+  }
+  uint64_t previous_timestamp_ns = previous_timestamp_ns_[function_address];
```

Comment:
We had the discussion at a couple of places, and always concluded to use `.at()` over `[]` for reading.
Currently, it is clear that the function is in the map and `[]` has no side-effect, but that might change it the future.

Maybe we should add this to our guidelines (do we actually have them yet?).

Reply:
Done.

Reply:
Good idea to add stuff like that into our coding guidelines. I think the only part we have here is this: https://github.com/google/orbit/blob/master/DEVELOPMENT.md#consistent-code-styling